codegen/masm/src/config.rs

```diff
@@ -26,8 +26,19 @@ pub struct CodegenConfig {
     pub ood_aux_frame_address: u64,
 
     // Memory range for the composition coefficients.
+    //
+    // The coefficients are organized as follows:
+    //
+    // 1. Transition coefficients for the main trace
+    // 2. Transition coefficients for the _aux_ trace
```

Comment:
> I would probably say "Transition constraint coefficients ..." instead of "Transition coefficients ...".

`Transition constraint coefficients` or `Integrity constraint coefficients`? I'm confused about the intended use.

> Do we need the underscore in front of _aux_trace? Or should it be just aux_trace?

I tend to skim over sentences that have a very similar words and miss the differences among them. To make this less likely, I usually introduce intentional differences to caught the eye, in this case I just went with a slated emphasis. I can remove if you want.

Reply:
The above is largely correct. There are some things we have in Winterfell which are not supported in AirScript, but there are also things in AirScript which are not supported by Winterfell.

On boundary constraints part you are completely correct: in AirScript we have boundary constraints against first/last rows only. In Winterfell we can describe much more general boundary constraints. (`Assertions` is just a way in which Winterfell describes boundary constraints).

On transition/integrity constraints we have the following situation:
- In AirScript, we have "integrity constraints", these break into two categories:
  - Transition constraints: constraints which describe a valid transition between a current an the next step.
  - Validity constraints (in my comment, I incorrectly called them "consistency constraints"): these constraints apply only to a single row. For example: $a^2 - a = 0$. Is a validity constraint because it enforces that all values in column $a$ must be binary.
- In Winterfell, there is no distinction between transition and validity constraints: everything is a transition constraint (even $a^2 - a = 0$ is described as a transition constraint). It works because we can describe a validity constraint with transition constraint + boundary constraint. But it is kind of hacky, and we'll probably change it in the future.

Reply:
changed:

- `1. Transition coefficients for the main trace` -> `Transition constraint coefficients for the main trace`.
- removed underline on `_aux_`